AK Apolo/app/src/main/java/ayds/apolo/songinfo/moredetails/model/repository/CardRepository.kt

```diff
@@ -1,43 +1,38 @@
 package ayds.apolo.songinfo.moredetails.model.repository
 
 import android.util.Log
+import ayds.apolo.songinfo.moredetails.model.broker.Broker
 import ayds.apolo.songinfo.moredetails.model.entities.*
-import ayds.apolo2.LastFM.LastFMAPIArtistService
-import ayds.apolo.songinfo.moredetails.model.repository.localLastFM.CardLocalStorage
-import ayds.apolo.songinfo.moredetails.utils.ArtistInfoToCardMapper
+import ayds.apolo.songinfo.moredetails.model.repository.localCard.CardLocalStorage
 
 interface CardRepository {
-    fun getCard(artistName: String): Card
+    fun getCards(artistName: String): List<Card>
 }
 
 internal class CardRepositoryImpl(
     private val cardDatabase: CardLocalStorage,
-    private val lastFMAPIArtistService: LastFMAPIArtistService,
-    private val artistInfoToCardMapper: ArtistInfoToCardMapper
+    private val broker: Broker
 ): CardRepository {
 
-    override fun getCard(artistName: String): Card {
-        var infoCard = cardDatabase.getCard(artistName)
+    override fun getCards(artistName: String): List<Card>{
+        var infoCardList = cardDatabase.getCards(artistName)
 
         when {
-            infoCard != null -> markCardAsLocal(infoCard)
-            else -> {
-                try {
-
-                    lastFMAPIArtistService.getArtist(artistName)?.let{
-                        infoCard = artistInfoToCardMapper.map(it)
-                        cardDatabase.saveCard(artistName, infoCard!!)
-                    }
-
-                } catch (e: Exception) {
-                    Log.e("Info Artist", "ERROR: $e")
-                }
+            infoCardList.isNotEmpty() -> markCardsAsLocal(infoCardList)
+            else -> try {
+                infoCardList = broker.getList(artistName)
+                cardDatabase.saveCards(artistName, infoCardList)
+            } catch (e: Exception){
+                Log.e("" , "ERROR: $e")
             }
         }
-        return infoCard ?: EmptyCard
+        return infoCardList
     }
 
-    private fun markCardAsLocal(card: Card) {
-        card.isLocallyStoraged = true
+    private fun markCardsAsLocal(list: List<Card>) {
+        for (card: Card? in list){
```

Comment:
Kotlin tiene varias extensiones para listas. Una es


```suggestion
       list.forEach{
           card?.isLocallyStoraged = true
       }
```

Además el nombre deberia ser cards, no list